staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -616,23 +616,67 @@ func (s *Scheme) generateConvertMeta(in interface{}) (conversion.FieldMatchingFl
 }
 
 // copyAndSetTargetKind performs a conditional copy before returning the object, or an error if copy was not successful.
-func copyAndSetTargetKind(copy bool, obj Object, kind schema.GroupVersionKind) (Object, error) {
+func (s *Scheme) copyAndSetTargetKind(copy bool, obj Object, kind schema.GroupVersionKind) (Object, error) {
 	if copy {
 		obj = obj.DeepCopyObject()
 	}
-	setTargetKind(obj, kind)
-	return obj, nil
+	err := s.setTargetKind(obj, kind)
+	return obj, err
```

Comment:
Please return `nil, err` if err is not nil, makes it clear that obj can't be used if there was an error.

Reply:
> Please return nil, err if err is not nil, makes it clear that obj can't be used if there was an error.

ok